### PR TITLE
Implement post filter

### PR DESCRIPTION
**Pull request by RossWebFS**
https://github.com/user-attachments/assets/37942db9-fecf-4112-933e-dbf2c3a9b8fa

### CODE REVIEW

File: src/pages/blog/BlogPage.tsx [hunk 1]
```diff
@@ -78,22 +26,38 @@ export default function BlogPage() {
...
+              columnsByBreakpoint={{
+                "0": 1,
+                "1000px": 2,
+              }}
+            >
+              {Array(postSkeletonCount)
+                .fill(undefined)
+                .map((_, index) => (
+                  <PostCard key={`skeleton-${index}`} post={undefined} />
+                ))}
```
Comment (mhevyk) [inline src/pages/blog/BlogPage.tsx:43]: Let's use `repeatComponent` util

**Comment by mhevyk** [comment]
As I can see from video, search is performed instantly when we type smth. I'd add debouncing logic for it

File: src/pages/blog/components/post-card/PostCard.tsx [hunk 2]
```diff
@@ -30,6 +30,8 @@ import {
 import LikeWidget from "@/pages/post/components/like-widget/LikeWidget";
 import theme from "@/theme/theme";
 
+// import { Post } from "@/api/generated";
```
[the remark this answers is not in the document]
Comment (RossWebFS) [reply src/pages/blog/components/post-card/PostCard.tsx:33]: Oh, thx for noticing! Removed

File: src/pages/blog/BlogPage.tsx (hunk 1 above)
Comment (RossWebFS) [reply src/pages/blog/BlogPage.tsx]: Replaced with repeatComponent function

**Review by mhevyk — COMMENTED** [review]
image

Eliminate setLike request

**Review by mhevyk — COMMENTED** [review]
image

What do you think about removinf search query param from url after we close search input?

**Review by mhevyk — COMMENTED** [review]
Can we tweak styling by adding decorations and so on? Io it is out of scope for this PR?

**Comment by RossWebFS** [comment]
> image
> 
> What do you think about removinf search query param from url after we close search input?

In my opinion, this might reduce the user experience because closing the search could trigger a layout re-render.

**Comment by RossWebFS** [comment]
> Can we tweak styling by adding decorations and so on? Io it is out of scope for this PR?

I think so. My plan was to create a separate pull request to fully complete the UI for this page

**Comment by RossWebFS** [comment]
> image
> 
> Eliminate setLike request

Fixed

**Review by mhevyk — APPROVED** [review]

File: src/pages/blog/BlogPage.tsx [hunk 3]
```diff
@@ -40,7 +40,8 @@ export default function BlogPage() {
             >
               {repeatComponent(<PostCardSkeleton />, postSkeletonCount)}
             </Masonry>
-          ) : data?.posts.length ? (
+          )}
+          {!loading && data?.posts.length && data.posts.length > 0 && (
```
Comment (mhevyk) [inline src/pages/blog/BlogPage.tsx:44]: Avoid using complex conditions inside jsx. Move it to variable `shouldRenderMasonry`

File: src/pages/blog/BlogPage.tsx [hunk 4]
```diff
@@ -40,7 +40,8 @@ export default function BlogPage() {
             >
```
Comment (mhevyk) [inline src/pages/blog/BlogPage.tsx:40]: Can't we just render `Masonry` component once and conditionally render skeleton here?

File: src/pages/blog/BlogPage.tsx [hunk 5]
```diff
@@ -52,9 +53,8 @@ export default function BlogPage() {
                 <PostCard key={post.id} post={post} />
               ))}
             </Masonry>
-          ) : (
-            <NoResults />
           )}
+          {!loading && !data?.posts.length && <NoResults />}
```
Comment (mhevyk) [inline src/pages/blog/BlogPage.tsx:57]: Same here

File: src/pages/blog/components/no-results/NoResults.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import Typography from "@mui/material/Typography";
+
+import { BoxStyled } from "@/pages/blog/components/no-results/NoResults.styled";
+
+export default function NoResults() {
+  return (
+    <BoxStyled>
+      {/* TODO: replace Typography with AppTypography and fix ui */}
```
Comment (mhevyk) [inline src/pages/blog/components/no-results/NoResults.tsx:8]: Job job! And when do we plan to replace it?

File: src/pages/blog/BlogPage.tsx (hunk 3 above)
Comment (RossWebFS) [reply src/pages/blog/BlogPage.tsx]: Fixed

image

File: src/pages/blog/BlogPage.tsx (hunk 4 above)
Comment (RossWebFS) [reply src/pages/blog/BlogPage.tsx]: Fixed

File: src/pages/blog/BlogPage.tsx (hunk 5 above)
Comment (RossWebFS) [reply src/pages/blog/BlogPage.tsx]: Fixed

File: src/pages/blog/components/no-results/NoResults.tsx (hunk 6 above)
Comment (RossWebFS) [reply src/pages/blog/components/no-results/NoResults.tsx]: Replaced with AppTypography

File: src/pages/blog/BlogPage.tsx [hunk 7]
```diff
@@ -78,22 +33,19 @@ export default function BlogPage() {
...
-        </BlogTabLayout>
-        <Box display="flex" justifyContent="center">
+        <BlogTabLayout />
+        <PostBox>
           <Masonry
             gap="2rem"
             columnsByBreakpoint={{
               "0": 1,
-              "1000px": 2,
+              "1000px": data && data.posts.length > 1 ? 2 : 1,
```
Comment (mhevyk) [inline src/pages/blog/BlogPage.tsx:42]: Try to avoid magic numbers

**Review by mhevyk — APPROVED** [review]

**Pull request merged**